Mu2eG4/geomMacros/README.md

```diff
@@ -0,0 +1,44 @@
+# Geometry Macros
+
+This directory contains some scripts that are useful for inspecting the geometry.
+
+## ```where_is_volume.C```
+This macro takes a string and will print all volumes that have that string in the name. It can take three arguments:
+* ```search_name``` is the name you are searching for (required)
+* ```gdmlname``` is the name of the GDML file you want to inspect (optional, default = ```mu2e.gdml```)
+* ```verbose``` is a boolean to print additional information about the volume (optional, default = ```false```)
+
+Examples:
+```
+root -l -b -q where_is_volume.C\(\"VirtualDetector\"\)
+root -l -b -q where_is_volume.C\(\"VirtualDetector\"\,\"mu2e.gdml\")
+root -l -b -q where_is_volume.C\(\"VirtualDetector\"\,\"mu2e.gdml\",true)
+```
+
+## ```find_volume_at_point.C```
+This macro takes an x, y, z position (in Mu2e global coordinates) and prints information for the volume at that position. Some warnings:
+* this will return the deepest volume in the hierarchy, and
+* if two volumes are touching at that point, the volume returned is ambiguous
```

Comment:
Abput bullet 1. If a volume A is fully contained inside volume B and you ask for a point on the boundary of volume A, round off errors could give you volume B instead of the expected volume A.

Instead of "ambiguous" maybe "apparently random".  I think that the real answer is that it will take the first match but the sort order might involve memory addresses so it will have randomness.

Also remind people that many StepPointMCs are on volume boundaries.

These ambiguities are why we invented the PhysicalVolumeInfo system but we never completed it properly.

Reply:
Thanks, Rob. Done in f540dd1ef

Reply:
Thanks Andy.